packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,465 @@
+			// do nothing, initialization not completed yet
+			// don't update brushHandle to avoid flash
+			return;
+		}
+		const { cartesianScales } = this.services;
+		const mainXScale = cartesianScales.getMainXScale();
+		const xScale = mainXScale.copy();
+
+		// get axes margins
+		let axesLeftMargin = 0;
+		const axesMargins = this.model.get("axesMargins");
+		if (axesMargins && axesMargins.left) {
+			axesLeftMargin = axesMargins.left;
+		}
+
+		this.overflowMenuStart = width - Configuration.toolBar.iconSize;
+		this.zoomOutStart =
+			this.overflowMenuStart - Configuration.toolBar.iconSize;
+		this.zoomInStart = this.zoomOutStart - Configuration.toolBar.iconSize;
+
+		const container = DOMUtils.appendOrSelect(svg, "svg.toolbar-container")
+			.attr("width", "100%")
+			.attr("height", Configuration.toolBar.height)
+			.attr("opacity", 1);
+
+		// clean children first
+		container.html(null);
+		if (isDataLoading) {
+			// put an empty rect to keep space unchanged
+			DOMUtils.appendOrSelect(container, "svg.toolbar-loading-space")
+				.append("rect")
+				.attr("height", Configuration.toolBar.height)
+				.attr("width", 1) // value doesn't matter but can't be empty
+				.attr("opacity", 0);
+		} else {
+			const self = this;
+
+			// zoom in icon and event
+			const zoomInContainer = DOMUtils.appendOrSelect(
+				container,
+				"svg.toolbar-zoomIn"
+			);
+			zoomInContainer.html(this.getZoomInIcon());
+
+			zoomInContainer.on("click", function () {
+				self.handleZoomInEvent(
+					self,
+					zoomDomain,
+					axesLeftMargin,
+					width,
+					xScale
+				);
+			});
+
+			// zoom out icon and event
+			const zoomOutContainer = DOMUtils.appendOrSelect(
+				container,
+				"svg.toolbar-zoomOut"
+			);
+			zoomOutContainer.html(this.getZoomOutIcon());
+
+			zoomOutContainer.on("click", function () {
+				self.handleZoomOutEvent(
+					self,
+					zoomDomain,
+					axesLeftMargin,
+					width,
+					xScale
+				);
+			});
+
+			// overflow menu icon and event
+			const overflowMenuContainer = DOMUtils.appendOrSelect(
+				container,
+				"svg.toolbar-overflow-menu"
+			);
+			overflowMenuContainer.html(this.getOverflowMenuIcon());
+			this.overflowMenuIcon = DOMUtils.appendOrSelect(
+				overflowMenuContainer,
+				`rect.${this.overflowIconClass}`
+			);
+
+			overflowMenuContainer.on("click", function () {
+				self.handleOverflowMenuEvent(
+					this.parentNode,
+					self,
+					axesLeftMargin,
+					width
+				);
+			});
+
+			document.body.addEventListener("click", function () {
+				if (
+					self.overflowMenuOptions
+						.selectAll("ul.bx--overflow-menu-options--open")
+						.size() > 0
+				) {
+					self.overflowIconClass = "icon-overflowRect";
+					self.overflowMenuIcon.attr("class", self.overflowIconClass);
+					self.services.events.dispatchEvent(Events.Toolbar.HIDE);
+				}
+			});
+
+			this.overflowIconClass =
+				this.overflowMenuOptions
+					.selectAll("ul.bx--overflow-menu-options--open")
+					.size() > 0
```

Comment:
```suggestion
				hasOpenedOverflowMenuOptions
```
use created variable for duplicate code in this block.